### ISSUE TITLE
[QUESTION] EOH in CEDS emissions

### ISSUE THREAD

**Issue by jennyfisher** [geoschem/geos-chem#643]
Hi GCST,

I'm using v12.8.0 and encountering an issue that I know disappears in v13.0.0 with the retirement of carbon-based units. For various reasons I don't want to upgrade for this project, so hoping one of you can help me clarify. Probably a question for @msulprizio or @christophkeller.

In this version, HEMCO reads in MOH and EOH emissions as follows:
```
# NOTE: EOH files in CEDS/v2018-08 are actually VOC1 (total alchohols) and are split into MOH, EOH, ROH here
0 CEDS_MOH_AGR    $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc           EOH_agr           1750-2014/1-12/1/0 C xy kg/m2/s MOH   26/90     1 5
0 CEDS_EOH_AGR    -                                                                    -                 -                  - -  -       EOH   26/91     1 5
...
# Convert CEDS total alchohols to methanol, ethanol, and other alcohols following Chen et al. (2019, ACP)
90 VOC1toMOH 0.5   - - - xy 1 1
91 VOC1toEOH 0.375 - - - xy 1 1
```

I've had a look at the Chen paper and confirmed those are mass ratios. This means the ratio of EOH/MOH emissions should be 37.5/50 = 0.75.

However, this isn't what comes out in the HEMCO diagnostics file. In fact, EOH emissions > MOH emissions (I've confirmed this was the case in the pre-v13.0.0 benchmarks, too).

I'm pretty sure that the issue is that the CEDS total alcohols files have an assumed molecular weight of 46 g/mol (aka ethanol MW) but in these older versions, EOH was carried in kgC (MW = 24). So I think the ratio I'm seeing in the output is (37.5*46/24)/50 = 1.4.

**My question is this:** Is this just a problem in the diagnostics -- i.e. HEMCO is accurately reading EOH as kg/m2/s and converting to kgC/m2/s for use in the model, but then messing up the conversion back to kg/m2/s when it saves to diagnostic? Or is this a problem at the start -- i.e. HEMCO is assuming the EOH emissions are already in kgC/m2/s and so is effectively ~doubling the emissions in the model (but then the conversion for the diagnostics is ok)?

And in either case -- what's the easiest way to resolve this in my simulations? If the latter is the problem, I assume I could just add a scale factor to the HEMCO_Config.rc, right?

Thanks!
Jenny

**Comment by msulprizio** [comment]
These differences were also noticeable during the GEOS-Chem 13.0.0 benchmarks, where we did the carbon-based unit removal. From the [1-month benchmark assessment form](http://wiki.seas.harvard.edu/geos-chem/index.php/GEOS-Chem_13.0.0#Benchmark_assessment_form):

>GEOS-Chem and HEMCO output in 12.9.0 and earlier indicated units of v/v or kg/m2/s for many of these species when it was actually vC/v and kgC/m2/s, and this led to incorrect unit conversions in GCPy when creating benchmark plots and tables. Where possible, we have corrected that in this benchmark output for 13.0.0 so we are comparing similar units for 12.9.0 and 13.0.0 output but some species like EOH still show large differences.

I think I mainly attributed it to a diagnostic issue at the time, but can't seem to locate my notes from that time. I'm running a quick simulation with 12.8.0 now with Verbose and Warnings set to 3 in HEMCO_Config.rc to get more details on the handling of those emissions and will report back here.

**Comment by msulprizio** [comment]
Hi @jennyfisher. I've attached relevant snippets from my HEMCO.log file below and will walk through what I think is going on. 

First, HEMCO reads the total alcohol emissions (in this example stored in variable `EOH_agr` in the netCDF file) to the `CEDS_MOH_AGR` container first. The emissions are read in as kg/m2/s and no unit conversion is applied since MOH is not carried in units carbon in GEOS-Chem or HEMCO (i.e. it has `MolecRatio : 1.0` in the species database). This is indicated by the line  `Data was in units of kg m-2 s-1 - unit conversion factor is    1.00000000000000` below.

```
-------------------------------------------------------------------------------
 Processing container: CEDS_MOH_AGR
 Parsing source file and replacing tokens
 Opening file: /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/CEDS/v2018-08/2014/EOH-em-anthro_CMIP_CEDS_2014.nc
HEMCO: Entering GET_TIMEIDX (hco_read_std_mod.F90) ( 6)
 Number of time slices found:           12
 Time slice range :    201401010000.000        201412010000.000
          preferred datetime:  201407010000.
              selected tidx1:              7
    corresponding datetime 1:  201407010000.
        assigned delta t [h]:              0
 local time?  F
HEMCO: Leaving GET_TIMEIDX (hco_read_std_mod.F90) ( 6)
 Reading variable EOH_agr
HEMCO WARNING: File units do not match: kg m-2 s-1 vs. kg/m2/s. File: /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/CEDS/v2018-08/2014/EOH-em-anthro_CMIP_CEDS_2014.nc
--> LOCATION: HCOIO_READ_STD (hcoio_read_std_mod.F90)
 Unit conversion settings:
 - Species MW         :    32.0400000000000
 - emitted compound MW:    32.0400000000000
 - molecular ratio    :    1.00000000000000
 - keep input species :  F
 - Year, month        :         2014           7
 Data was in units of kg m-2 s-1 - unit conversion factor is    1.00000000000000
   ==> Use map_a2a regridding
HEMCO: Leaving HCOIO_READ_STD (hcoio_read_std_mod.F90) ( 5)
HEMCO: Leaving HCOIO_DataRead (hcoio_dataread_mod.F90) ( 4)
HEMCO: Entering tIDx_Assign (hco_tidx_mod.F90) ( 4)
HEMCO: Leaving tIDx_Assign (hco_tidx_mod.F90) ( 4)
HEMCO: Entering EmisList_Pass (hco_emislist_mod.F90) ( 4)
HEMCO: Entering EmisList_Add (HCO_EMISLL_MOD.F90) ( 5)
HEMCO: Entering Add2EmisList ( 6)
HEMCO: Leaving Add2EmisList ( 6)
Container added to EmisList:
Container CEDS_MOH_AGR
    -->Data type       :            1
    -->Container ID    :          647
    -->Target ID       :          647
    -->File data home?              1
    -->Source file     : $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc
    -->ncRead?            T
    -->Shared data file?  F
    -->Source parameter: EOH_agr
    -->Year range      :         1750        2014
    -->Month range     :            1          12
    -->Day range       :            1           1
    -->Hour range      :            0           0
    -->SpaceDim        :            2
    -->Array dimension :           72          46
    -->Array sum       :   0.0000000E+00
    -->Array min & max :   0.0000000E+00  0.0000000E+00
    -->Time dimension  :            1
    -->Delta t[h]      :            0
    -->Local time?        F
    -->Tempres         : Constant
    -->OrigUnit        : kg/m2/s
    -->Concentration?     F
    -->Coverage        :         -999
    -->Extension Nr    :            0
    -->Species name    : MOH
    -->HEMCO species ID:          144
    -->Category        :            1
    -->Hierarchy       :            5
    -->2D emitted into :    1.00000000000000       and    1.00000000000000
HEMCO: Leaving EmisList_Add (HCO_EMISLL_MOD.F90) ( 5)
HEMCO: Leaving EmisList_Pass (hco_emislist_mod.F90) ( 4)
HEMCO: Entering tIDx_Assign (hco_tidx_mod.F90) ( 4)
HEMCO: Leaving tIDx_Assign (hco_tidx_mod.F90) ( 4)
HEMCO: Entering EmisList_Pass (hco_emislist_mod.F90) ( 4)
HEMCO: Entering EmisList_Add (HCO_EMISLL_MOD.F90) ( 5)
HEMCO: Entering Add2EmisList ( 6)
HEMCO: Leaving Add2EmisList ( 6)
```

Next, HEMCO adds the container `CEDS_EOH_AGR` using the same input just read. There is apparently no unit conversion factor applied here. HEMCO must assume the unit conversion applied when the variable is first read applies to the following containers that are scaled from that field.

```
Container added to EmisList:
Container CEDS_EOH_AGR
    -->Data type       :            1
    -->Container ID    :          648
    -->Target ID       :          648
    -->File data home?              0
    -->Source file     : $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc
    -->ncRead?            T
    -->Shared data file?  T
    -->Source parameter: EOH_agr
    -->Year range      :         1750        2014
    -->Month range     :            1          12
    -->Day range       :            1           1
    -->Hour range      :            0           0
    -->SpaceDim        :            2
    -->Array dimension :           72          46
    -->Array sum       :   0.0000000E+00
    -->Array min & max :   0.0000000E+00  0.0000000E+00
    -->Time dimension  :            1
    -->Delta t[h]      :            0
    -->Local time?        F
    -->Tempres         : Constant
    -->OrigUnit        : kg/m2/s
    -->Concentration?     F
    -->Coverage        :         -999
    -->Extension Nr    :            0
    -->Species name    : EOH
    -->HEMCO species ID:           59
    -->Category        :            1
    -->Hierarchy       :            5
    -->2D emitted into :    1.00000000000000       and    1.00000000000000
HEMCO: Leaving EmisList_Add (HCO_EMISLL_MOD.F90) ( 5)
HEMCO: Leaving EmisList_Pass (hco_emislist_mod.F90) ( 4)
HEMCO: Entering HCOIO_DataRead (hcoio_dataread_mod.F90) ( 4)
HEMCO: Entering HCOIO_READ_STD (hcoio_read_std_mod.F90) ( 5)
-------------------------------------------------------------------------------
```

Later, when HEMCO computes the emissions it applies the scale factors specified in the `CEDS_EOH_AGR` entry in HEMCO_Config.rc. This includes `VOC1toEOH` or ScalID 91.

```
-------------------------------------------------------------------------------
 Calculating emissions for species EOH
-------------------------------------------------------------------------------
HEMCO: Entering GET_CURRENT_EMISSIONS ( 3)
                                                                               
 Evaluate field CEDS_EOH_AGR
Applying scale factor GEIA_TOD_FOSSIL
 Scale field GEIA_TOD_FOSSIL
 Time slice:           21
 IX, IY:           25          25
 Scale factor (IX,IY,L1):   0.6500000
 Mathematical operation :            1
Applying scale factor VOC1toEOH
 Scale field VOC1toEOH
 Time slice:            1
 IX, IY:           25          25
 Scale factor (IX,IY,L1):   0.3750000
 Mathematical operation :            1
HEMCO: Leaving GET_CURRENT_EMISSIONS ( 3)
Successfully updated diagnostics: InvCEDS_EOH (counter:  1)
HEMCO: Entering GET_CURRENT_EMISSIONS ( 3)
```

To see if I could get HEMCO to automatically recognize EOH should be carried in kgC/m2/s (in this version of HEMCO), I tried removing the dashes from the `CEDS_EOH_AGR` entry:

```
0 CEDS_MOH_AGR    $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc           EOH_agr           1750-2014/1-12/1/0 C xy kg/m2/s MOH   26/90     1 5
0 CEDS_EOH_AGR    $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc           EOH_agr           1750-2014/1-12/1/0 C xy kgC/m2/s EOH   26/91     1 5
```

Sure enough HEMCO does actually convert from kg/m2/s to kgC/m2/s in this case:

```
-------------------------------------------------------------------------------
 Processing container: CEDS_EOH_AGR
 Parsing source file and replacing tokens
 Reading from existing stream: /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/CEDS/v2018-08/2014/EOH-em-anthro_CMIP_CEDS_2014.nc
HEMCO: Entering GET_TIMEIDX (hco_read_std_mod.F90) ( 6)
 Number of time slices found:           12
 Time slice range :    201401010000.000        201412010000.000
          preferred datetime:  201407010000.
              selected tidx1:              7
    corresponding datetime 1:  201407010000.
        assigned delta t [h]:              0
 local time?  F
HEMCO: Leaving GET_TIMEIDX (hco_read_std_mod.F90) ( 6)
 Reading variable EOH_agr
HEMCO WARNING: File units do not match: kg m-2 s-1 vs. kgC/m2/s. File: /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/CEDS/v2018-08/2014/EOH-em-anthro_CMIP_CEDS_2014.nc
--> LOCATION: HCOIO_READ_STD (hcoio_read_std_mod.F90)
 Unit conversion settings:
 - Species MW         :    46.0700000000000
 - emitted compound MW:    12.0000000000000
 - molecular ratio    :    2.00000000000000
 - keep input species :  F
 - Year, month        :         2014           7
 Data was in units of kg m-2 s-1 - converted to HEMCO units by applying scale factor   0.520946385934448
```

In terms of the diagnostic output, HEMCO appears to apply a mass unit conversion factor depending on the output units specified in HEMCO_Diagn.rc. I added print statements to hco_diagn_mod.F90 to determine if/when scaling was being applied to diagnostic output:

```
          !----------------------------------------------------------------
          ! Scale factor for mass. This determines the scale factor from
          ! HEMCO mass unit (kg) to the desired output unit.
          ! HCO_UNIT_MassCal returns the mass scale factor from OutUnit to
          ! HEMCO unit, hence need to invert this value!
          !----------------------------------------------------------------
          IF ( .NOT. PRESENT(MW_g)       .OR. &
               .NOT. PRESENT(EmMW_g)     .OR. &
               .NOT. PRESENT(MolecRatio)       ) THEN
             MWg   = HcoState%Spc(HcoID)%MW_g
             EmMWg = HcoState%Spc(HcoID)%EmMW_g
             MolR  = HcoState%Spc(HcoID)%MolecRatio
          ELSE
                MWg   = MW_g
                EmMWg = EmMW_g
                MolR  = MolecRatio
          ENDIF

          CALL HCO_UNIT_GetMassScal( HcoState%Config,             &
                    unt         = OutUnit,                        &
                    MW_IN       = MWg,                            &
                    MW_OUT      = EmMWg,                          &
                    MOLEC_RATIO = MolR,                           &
                    Scal        = Scal                             )
          # MPS : ADDED DEBUG MSG HERE #
          WRITE(MSG,*) 'Diagnostic mass scale factor : ', Scal
          CALL HCO_MSG(HcoState%Config%Err,MSG)
          IF ( Scal <= 0.0_hp ) THEN
             MSG = 'Cannot find mass scale factor for unit '//TRIM(OutUnit)
             CALL HCO_ERROR( HcoState%config%Err, MSG, RC, THISLOC=LOC )
             RETURN
          ENDIF
          ThisDiagn%MassScal = 1.0_hp / Scal
```

When the output units of EOH emissions in HEMCO_Diagn.rc are listed as kg/m2/s, it returns:

```
 Diagnostic mass scale factor :   0.520946385934448
 Diagnostic area scale factor :    1.00000000000000
Successfully added diagnostic InvCEDS_EOH to collection    1
```

but when they're listed as kgC/m2/s, it returns:

```
 Diagnostic mass scale factor :    1.00000000000000
 Diagnostic area scale factor :    1.00000000000000
Successfully added diagnostic InvCEDS_EOH to collection    1
```

To summarize, in the default setup it appears that HEMCO is reading in the EOH emissions in kg/m2/s and not properly converting to kgC/m2/s. When saving the diagnostic output to the default units of kg/m2/s, HEMCO divides by the mass scaling factor 0.52 even though units are already in kg/m2/s. To resolve this you can (1) remove the dashes from EOH entries in HEMCO_Config.rc to ensure units are properly converted to kgC/m2/s and (2) change output units in HEMCO_Diagn.rc to kgC/m2/s.

**Comment by jennyfisher** [comment]
Thank you for all this Melissa! I have a couple of follow-up questions (and one potential concern...).

The first is just a clarification - is this the suggested format of the line to use in HEMCO_Config.rc?
`0 CEDS_EOH_AGR    $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc           EOH_agr           1750-2014/1-12/1/0 C xy kgC/m2/s EOH   26/91     1 5`

Specifically -- do I put `kgC/m2/s` on that line (rather than `kg/m2/s`)? I get a bit confused here as the data in the file are in kg/m2/s, but kgC/m2/s is what the model needs (right?).

Second, I'm wondering if there will be errors in the MOH line (and these might translate to the current version of the model, too). Here's how that line looks currently:
`0 CEDS_MOH_AGR    $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc           EOH_agr           1750-2014/1-12/1/0 C xy kg/m2/s MOH   26/90     1 5`

As I understand it, this means HEMCO will read in the `EOH_agr` field, which is ethanol, and apply it to MOH with scaling factor 90. In the CEDS file metadata, it specifies the molecular weight is 46, the molecular weight of ethanol (EOH) not methanol (MOH). Does another scaling factor need to be applied to MOH to take into account the molecular weight differences between the species? Or is this not necessary because this is accounted for in scaling factor 90 (which I think is just saying 50% of the total mass of that input field should go into MOH). I'm a bit confused by these scale factors...

Apologies if I'm overcomplicating things here!!

Cheers,
Jenny

**Comment by msulprizio** [comment]
Hi @jennyfisher. You would want to list `kg/m2/s` for the `CEDS_EOH_*` entries. The units listed there should match the units in the netCDF file and if needed HEMCO will convert to the internal units of kg/m2/s or kgC/m2/s depending on the species. Otherwise, yes, your `CEDS_EOH_AGR` entry above looks correct.

For MOH, I'm not sure if current scaling factors account for total mass or not. This may be a question better suited for Dylan Millet or Xin Chen, as they implemented methanol in GEOS-Chem. I'll contact them and follow up here.

**Comment by jennyfisher** [comment]
Thanks Melissa. I'm still a little confused though! I also though I would list as `kg/m2/s` to match what is in the netCDF file. But in your example above, you showed that the conversion only worked properly when you set up the input file line as:

`0 CEDS_EOH_AGR    $ROOT/CEDS/v2018-08/$YYYY/EOH-em-anthro_CMIP_CEDS_$YYYY.nc           EOH_agr           1750-2014/1-12/1/0 C xy kgC/m2/s EOH   26/91     1 5
`

Which gave this output in the HEMCO.log:
```
 Reading variable EOH_agr
HEMCO WARNING: File units do not match: kg m-2 s-1 vs. kgC/m2/s. File: /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/CEDS/v2018-08/2014/EOH-em-anthro_CMIP_CEDS_2014.nc
--> LOCATION: HCOIO_READ_STD (hcoio_read_std_mod.F90)
 Unit conversion settings:
 - Species MW         :    46.0700000000000
 - emitted compound MW:    12.0000000000000
 - molecular ratio    :    2.00000000000000
 - keep input species :  F
 - Year, month        :         2014           7
 Data was in units of kg m-2 s-1 - converted to HEMCO units by applying scale factor   0.520946385934448
```

Was that just a typo in the first line, or was that `kgC/m2/s` necessary to get HEMCO to apply the required scale factor?

Thanks,
Jenny

**Comment by msulprizio** [comment]
Hi @jennyfisher. It turns out it doesn't matter if you list `kg/m2/s` or `'kgC/m2/s`. The units entry in HEMCO_Config.rc is mainly used to check what the user thinks are the units against what is actually in the netCDF file. HEMCO will do a comparison and if the unit strings don't match HEMCO will either print a warning to HEMCO.log or stop with an error -- this is controlled by the `Unit tolerance` option at the top of HEMCO_Config.rc. 

I ran a quick simulation using the unit string `kg/m2/s` for `CEDS_EOH_AGR` and got:
```
 Reading variable EOH_agr
HEMCO WARNING: File units do not match: kg m-2 s-1 vs. kg/m2/s. File: /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/CEDS/v2018-08/2014/EOH-em-anthro_CMIP_CEDS_2014.nc
--> LOCATION: HCOIO_READ_STD (hcoio_read_std_mod.F90)
 Unit conversion settings:
 - Species MW         :    46.0700000000000
 - emitted compound MW:    12.0000000000000
 - molecular ratio    :    2.00000000000000
 - keep input species :  F
 - Year, month        :         2014           7
 Data was in units of kg m-2 s-1 - converted to HEMCO units by applying scale factor   0.520946385934448
```

Then I ran a simulation using `kgC/m2/s` for that field and got:
```
 Reading variable EOH_agr
HEMCO WARNING: File units do not match: kg m-2 s-1 vs. kgC/m2/s. File: /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/CEDS/v2018-08/2014/EOH-em-anthro_CMIP_CEDS_2014.nc
--> LOCATION: HCOIO_READ_STD (hcoio_read_std_mod.F90)
 Unit conversion settings:
 - Species MW         :    46.0700000000000
 - emitted compound MW:    12.0000000000000
 - molecular ratio    :    2.00000000000000
 - keep input species :  F
 - Year, month        :         2014           7
 Data was in units of kg m-2 s-1 - converted to HEMCO units by applying scale factor   0.520946385934448
```

As you can see, the same unit conversion factor (0.52) is applied in both cases because HEMCO recognizes the units in the netCDF files are kg m-2 s-1 and automatically converts to kgC/m2/s for species EOH.

**Comment by jennyfisher** [comment]
Fantastic, thanks!

**Comment by msulprizio** [comment]
I'm going to close out this issue for now, but you may always request for us to reopen or just submit a new issue if there are additional questions.

**Comment by jennyfisher** [comment]
Thanks Melissa - was just waiting on an answer from Xin about the scaling factors... I'm sure that can get added separately when we have a response!

**Comment by msulprizio** [comment]
Thanks for the reminder! Yes, we can add the response here when we hear back.

**Comment by msulprizio** [comment]
Xin Chen wrote:

>**Regarding the ethanol unit conversion:** I agree with Melissa that the EOH line in HEMCO_Config.rc should have a unit of kgC/m2/s to allow the unit conversion to happen at the beginning. This was properly set in Chen et al. 2019 paper simulations which used other global anthropogenic inventories, but was not carefully checked when delivering to GCST as that delivery was specifically created for methanol. Sorry about that!
>
>**Regarding MOH**, despite the molecular weight (MW) of 46.2 (which equals to the MW of ethanol) in CEDS file, I was interpreting it as total alcohols based on the file names (looks like the original file name is "VOC01-alcohols-em-speciated-VOC-anthro_input4MIPs_emissions_CMIP_CEDS-2017-05-18-supplemental-data_gn_200001-201412.nc" and the original variable ID is "VOC01_alcohols_em_speciated_VOC_anthro").
>
>I'm not sure but I feel that the VOC speciation across different generations of inventories may follow somewhat the same speciation profile, with similar groupings and MW notations. For example, I see the same information in the old GEIA inventory page (GEIA link) and in the CEDS file (CEDS speciation readme). And we can see that the GEIA page lists both methanol and ethanol in the alcohols group which has an average MW of 46.2 (though a mixture of methanol + ethanol should have MW less than 46).
>
>So I ended up interpreting it as total alcohols, and assume that the 46.2 in the file is the average molecular weight of a mixture of MOH (MW<46), EOH (MW=46), and >C2 alcohols (MW>46), and a simple partitioning: 50% MOH, 37.5% EOH, and 12.5% other alcohols except MOH and EOH. The 50% MOH assumption followed Dylan and Kelley's old methanol work (not documented in their papers though).
>
>To summarize, yes, the scale factors currently applied to convert total alcohols to MOH are for the total mass. I don't think we need to apply additional scaling, unless there's a document clearly say that the "alcohols" in those anthropogenic inventories refer to "pure ethanol".
